### DISCUSSION TITLE
.net Core

### DISCUSSION THREAD

**Discussion by icavalheiro** [NoCode-NoLife/melia#168] [Discussions]
I'm following this project for a while now and I decided that I want to contribute a little bit with it but working with mono is really boring and a big loss of performance when running under unix systems... 
Is there any intention on porting this project to .net core? 

I would be able to do the port, as well as setting up docker environments and also contributing with the development of the server (got 10 years of exp in game development).

**Comment by exectails** [comment]
Hi, I'm happy to see someone wanting to contribute. I haven't actively considered porting Melia to Core yet, because I haven't done anything with Core in general. How much effort would porting it be? Running the Portability Analyzer it seems that Swebs, our web server library, isn't fully compatible yet, neither are other libraries we use, though I do assume there would be replacements.

**Comment by icavalheiro** [comment]
Its hard to say how long it could takes, specially becouse I don't know the project that well. But besides uncompatible libraries it should be as ease as changing the target from VS. But for the libraries we would have to replace all references to libs that do not exist in the dotnet core environment. And also, I did see that you do not use migrations/EF to handle DB, this is definately something to look forward during the transition process, since EF and migrations are much more reliable/productive than executing SQL queries and saving migrations as SQL files.

I'll have to take a few days to get used to the project before I can come up with any estimates.

**Comment by exectails** [comment]
Closing this discussion since we've switched to .NET 8 by this point.
